src/res_scanner.ml

```diff
@@ -487,11 +487,11 @@ let scanTemplateLiteralToken scanner =
       | _ ->
         next scanner;
         scan ())
-    | ch when ch == hackyEOFChar ->
+    | ch when ch = hackyEOFChar ->
```

Comment:
Why do you prefer `=` here?

Reply:
`==` is not correct here, it happens to work since char is a immediate number, if it is float which is boxed in ocaml native this will be broken